lib/Sema/CSSimplify.cpp

```diff
@@ -2423,9 +2423,14 @@ ConstraintSystem::matchTypes(Type type1, Type type2, ConstraintKind kind,
         conversionsOrFixes.push_back(Fix::getForcedDowncast(*this, type2));
     }
 
-    // If we're converting an lvalue to an inout type, add the missing '&'.
-    if (type2->getRValueType()->is<InOutType>() && type1->is<LValueType>()) {
-      conversionsOrFixes.push_back(FixKind::AddressOf);
+    if (type2->getRValueType()->is<InOutType>()) {
```

Comment:
It's not valid to have an InOutType wrap an LValueType, so you can remove the call to getRValueType() here